src/output-json.c

```diff
@@ -705,6 +705,9 @@ void CreateJSONFlowId(json_t *js, const Flow *f)
     if (f->parent_id) {
         json_object_set_new(js, "parent_id", json_integer(f->parent_id));
     }
+    if (f->tenant_id > 0) {
```

Comment:
I don't think this belongs in this function. Probably the best place would be `JsonAddCommonOptions`.

I am a bit conflicted about whether its a good idea at all. I see why you'd want it, but at the same time the tentants are detection engine construct, not a global construct (at this time). So it really only has meaning in detection engine related output, which means alerts. Trying to think if we can just use it like this, or that we need to consider edge cases like can 2 sides of a flow have different tenants?

Reply:
> I am a bit conflicted about whether its a good idea at all. I see why you'd want it, but at the same time the tentants are detection engine construct, not a global construct (at this time).

As far as I have understood from the code the tenant-id is determined during receive time and passed along to the flow logic. The detect engine effectively pulls it out of the flow data when it detects something. Isn't the flow data central to all engines?

This is also the reason why I added this field in this function as this is the place where the `flow_id` gets added.

> Trying to think if we can just use it like this, or that we need to consider edge cases like can 2 sides of a flow have different tenants?

I haven't looked into the flow bookkeeping. If the fields that the multi-detect selections are based on (or the tenant_id themselves) are part of what makes up a unique flow then a flow would be unique. If this doesn't hold true then this is equally broken for the detection engine as half of the flow could have been received using vlan A and half over vlan B and the detection engine also only flags one of the two (probably using the tenant_id of the packet that instantiated the flow).

Reply:
@victorjulien: Do you see a way forward for this change?

I can look at moving this behind the "JsonAddCommonOptions" logic.

But I don't understand how the current logic works regarding the "the tentants are detection engine construct, not a global construct". We want to use it for things like http and dns eve logging and I'm trying to see how these event types would have different `tenant_id` values then alerts from the detection engine that use the http or dns decoder results.
Or put another way: what would it entail for this to be a global construct? Perhaps we can help with that.

Reply:
Other than in the case of processing a pcap using unix socket the tenants are only assigned in the detection engine. So if you'd run suri with `--disable-detection` it wouldn't assign them.

That said I don't have an objection to logging this when its set (meaning > 0). It would be good to add a comment to the docs to explain the dependency on the detect engine.

Reply:
Sorry for the delay, I've just returned from vacation. Thanks for the clarification, I'll also take a look at the necessary documentation changes.